orangecontrib/snom/widgets/preprocessors/linelevel.py

```diff
@@ -17,7 +17,13 @@ def __init__(self, method="median"):
         self.method = method
 
     def transform_image(self, image, data):
-        return LineLevel(method=self.method).transform(image)
+        if "datatype" in data.attributes:
```

Comment:
This is a complication. I would do:
`datatype = data.attributes.get("datatype", preferred_default_value)`

which would avoid the ifs.

Reply:
@markotoplak thanks for the advice! I'll do the modifications!